imap_processing/ultra/l0/decom_ultra.py

```diff
@@ -71,6 +65,9 @@ def append_tof_params(
 
     # Once "SID" reaches 7, we have all the images and data for the single timestamp
     if packet.data["SID"].derived_value == 7:
+        decom_data["SHCOARSE"].extend(list(set(data_dict["SHCOARSE"])))
+        data_dict["SHCOARSE"].clear()
```

Comment:
I don't know enough. Just asking out of curiosity. What is the reason for clearing SHCOARSE from this dictionary?

Reply:
data_dict is used to accumulate all of the tof images for a single timestamp. Once they are accumulated we use np.stack. Here is an example:

# Example data_dict with key 'key' having a list of arrays
data_dict = {
    'key': [np.array([1, 2, 3]), np.array([4, 5, 6]), np.array([7, 8, 9])]
}

# Stacking the arrays
stacked_array = np.stack(data_dict['key'])

array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])

I use data_dict for SHCOARSE just to accumulate the data and combine it when we have a complete dataset. Afterwards it is cleared. Notice that I have a TODO at the top of test_decom_apid_883.py:

# TODO: discuss with instrument team incomplete set of SIDs
I have this as a TODO because there might be cases in which the dataset is incomplete and so this part of the code will change if that is the case. But since I haven't talked with the instrument team yet I think I should leave it as is for now. 

Nice catch :)